sklearn/tree/_tree.pyx

```diff
@@ -109,16 +108,14 @@ cdef class TreeBuilder:
             # since we have to copy we will make it fortran for efficiency
             X = np.asfortranarray(X, dtype=DTYPE)
 
-        if y.dtype != DOUBLE or not y.flags.contiguous:
-            y = np.ascontiguousarray(y, dtype=DOUBLE)
```

Comment:
The reason for removing this is that this condition is already being checked in https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/tree/_classes.py#L247-L248.

Reply:
To improve readability of the changes, I would keep this PR focused on the scope of removing the `-Wcpp warnings` and suggest removing this check within another PR, WDYT?